tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_access_opened_files.py

```diff
@@ -0,0 +1,223 @@
+
+
+# Tests
+@pytest.mark.parametrize(
+    "operation, tags_to_apply", [("delete", {"ossec_conf"}), ("rename", {"ossec_conf"})]
+)
+def test_basic_usage_access_opened_files(
+    operation,
+    tags_to_apply,
+    get_configuration,
+    configure_environment,
+    create_and_delete_large_file,
+    restart_syscheckd_function,
+    wait_for_scan_start,
+):
+    """
+    description: Check that files that are being scanned by syscheckd daemon
+                 can modified (renamed/deleted), and that wazuh is not
+                 blocking the files.
+
+    wazuh_min_version: 4.3
+
+    parameters:
+        - operation:
+            type: string
+            brief: Tells which operation has to be performed.
+        - tags_to_apply:
+            type: string
+            brief: Tells which configuration to use.
+        - get_configuration:
+            type: fixture
+            brief: Get configurations from the module.
+        - configure_environment:
+            type: fixture
+            brief: Configure a custom environment for testing.
+        - create_and_delete_large_file:
+            type: fixture
+            brief: Creates file to be monitored. Cleans enviroment after test.
+        - restart_syscheckd_function:
+            type: fixture
+            brief: Restart the `wazuh-syscheckd` daemon.
+        - wait for scan start:
+            type: fixture
+            brief: Wait for start of initial FIM scan start.
+
+    assertions:
+        - Verify that the file hast been modified (renamed/deleted).
+        - Verify that the modificaction is done before the initial scan ends.
+
+    input_description: Two use cases are found in the test module and include
+                       parameters for operation (`delete` and `rename`).
+    """
+
+    check_apply_test(tags_to_apply, get_configuration["tags"])
+
+    # Wait a few seconds for scan to run on created file.
+    time.sleep(3)
+
+    modify_time = None
+    # Modify/Delete the file
+    if operation == "rename":
+        changed_path = os.path.join(directory_str, "changed_name")
+        try:
+            modify_time = time.time()
+            os.rename(file_path, changed_path)
+            # Assert the file has been changed
+            assert os.path.isfile(changed_path) == True
+        except (OSError, IOError, PermissionError) as error:
+            pytest.fail(f"Could not rename file - Error: {error}")
+    elif operation == "delete":
+        try:
+            os.remove(file_path)
+            modify_time = time.time()
+            # Assert the file has been deleted
+            assert os.path.isfile(file_path) == False
+        except (OSError, IOError, PermissionError) as error:
+            pytest.fail(f"Could not delete file - Error: {error}")
+
+    # Capture scan end timestamp & assert the file was modified before scan end
+    scan_timestamp = get_scan_timestamp(wazuh_log_monitor)
+    assert modify_time < scan_timestamp
```

Comment:
I think, that this test is better if to be split into 2:
 - One that is `renamed` file 
 - and one is `deleted` file.

 It would help to modularize the cases.

Reply:
Done